homeassistant/components/decora_ble/config_flow.py

```diff
@@ -0,0 +1,155 @@
+
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            result = await self._attempt_create_entry(address, user_input[CONF_NAME])
+            if isinstance(result, str):
+                errors["base"] = result
+            else:
+                return result
+
+        placeholders = {"name": self._discovery_info.name}
+        self.context["title_placeholders"] = placeholders
+
+        field_values = (
+            user_input if user_input is not None else {CONF_NAME: DEFAULT_NAME}
+        )
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_NAME, default=field_values.get(CONF_NAME)): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="bluetooth_confirm",
+            description_placeholders=placeholders,
+            data_schema=data_schema,
+            errors=errors,
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the user step to pick discovered device."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            address = user_input[CONF_ADDRESS]
+
+            await self.async_set_unique_id(address, raise_on_progress=False)
+            self._abort_if_unique_id_configured()
+
+            result = await self._attempt_create_entry(address, user_input[CONF_NAME])
+            if isinstance(result, str):
+                errors["base"] = result
+            else:
+                return result
+
+        current_addresses = self._async_current_ids()
+        for discovery in async_discovered_service_info(self.hass):
+            if (
+                discovery.address in current_addresses
+                or discovery.address in self._discovered_devices
+                or DECORA_SERVICE_UUID not in discovery.service_uuids
+            ):
+                continue
+            self._discovered_devices[discovery.address] = discovery
+
+        if not self._discovered_devices:
+            return self.async_abort(reason="no_devices_found")
+
+        field_values = (
+            user_input if user_input is not None else {CONF_NAME: DEFAULT_NAME}
+        )
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_ADDRESS): vol.In(
+                    {
+                        service_info.address: (
+                            f"{service_info.name} ({service_info.address})"
+                        )
+                        for service_info in self._discovered_devices.values()
+                    }
+                ),
+                vol.Required(CONF_NAME, default=field_values.get(CONF_NAME)): str,
+            }
+        )
+        return self.async_show_form(
+            step_id="user",
+            data_schema=data_schema,
+            errors=errors,
+        )
+
+    async def _attempt_create_entry(
+        self, address: str, name: str
+    ) -> Union[FlowResult, str]:
```

Comment:
Please avoid progressing the flow in the helper.

Make this return an error string or None and let the caller progress the flow based on that.